pkg/webhook/server.go

```diff
@@ -82,6 +82,9 @@ type Server struct {
 	// WebhookMux is the multiplexer that handles different webhooks.
 	WebhookMux *http.ServeMux
 
+	// WebhookWrapper is a custom wrapper func that enables adding middleware onto the webhook mux.
+	WebhookWrapper func(http.Handler) (http.Handler, error)
```

Comment:
Why not name it like what it is, namely a `Middleware`?

Also, what is the reason for the error return here? I realize this could error during request handling, but what would be a reason where it could error here?

Reply:
The error was just to make it flexible in case whatever setup function there is could fail and therefore bubble up the error to the Start() method. Note that this isn't error of request handling but error of setting up the request handler/middleware. 

Not sure if there's a common signature for middleware - hence why I went with wrapper but can change if there's agreement